README_cmake.md

```diff
@@ -0,0 +1,213 @@
+
+Then, using a Generator can be as simple as
+
+    halide_library(coolness SRCS coolness_generator.cpp)
+
+    add_executable(my_app my_app.cpp)
+    target_link_libraries(my_app PUBLIC coolness)
+
+
+## Build Rules
+
+```
+halide_library(name, srcs, extra_outputs, filter_deps, function_name, 
+generator_args, generator_deps, generator_name, halide_target, 
+halide_target_features, includes)
+```
+
+*   **name** *(Name; required)* The name of the library target.
+
+*   **srcs** *(List of C++ source files; required)* source file(s) to compile into the
+    Generator Executable.
+
+*   **extra_outputs** *(List of strings; optional)* A list of extra Halide
+    outputs to generate at build time; this is exclusively intended for
+    debugging (e.g. to examine Halide code generation) and currently supports:
+
+    *   "assembly" (generate assembly listings for the generated functions)
+    *   "bitcode" (emit the LLVM bitcode for generation functions)
+    *   "stmt" (generate Halide .stmt files for generated functions)
+    *   "html" (like "stmt", but generated with HTML-formatted wrapping)
+
+*   **filter_deps** *(List of CMake targets; optional)* optional list of dependencies
+    needed by the Filter.
+
+*   **function_name** *(String; optional)* The name of the generated C function
+    for the filter. If omitted, defaults to the CMake rule name.
+
+*   **generator_args** *(String; optional)* Arguments to pass to the Generator,
+    used to define the compile-time values of GeneratorParams defined by the
+    Generator. If any undefined GeneratorParam names (or illegal GeneratorParam
+    values) are specified, a compile error will result.
+
+*   **generator_deps** *(List of CMake targets; optional)* optional list of
+    dependencies needed by the Generator. (These dependencies are *not* included
+    in the filter produced by `halide_library`, nor in any executable that
+    depends on it.)
+
+*   **generator_name** *(String; optional)* The registered name of the Halide
+    generator (i.e., the name passed as the second argument to HALIDE_REGISTER_GENERATOR
+    in the Generator source file). If omitted, defaults to the CMake rule name.
+
+*   **halide_target** *(String; optional)* The Halide target-string to
+    use when generating code; if omitted, defaults to "host". If you want to target
+    for a specific architechture, or to enable certain features, you can specify
+    specific values, e.g. "x86-64-linux-sse41-avx" compiles for x86-64 on Linux, 
+    also enabling SSE4.1 and AVX. You can also specify multiple values, separated
+    by commas, to generate several variants that are selected at runtime; e.g.
+    "x86-64-linux-sse41-avx,x86-64-linux-sse41,x86-64-linux" will generate
+    three versions of the code (for SSE4.1+AVX, for SSE4.1, and for plain x86-64),
+    and select the first matching variants at runtime.
+
+*   **halide_target_features** *(List of strings; optional)* A list of extra
```

Comment:
Can we expand this a bit to mention this is where the GPU enable flags go. Possibly giving an example. I see it as a point of confusion because the CPU ISA specializations happen above due to multitarget (I think). (Though I suppose at some point we should do multi-target for different versions of CUDA and such...)

Reply:
Actually, also add a "more detail below pointer" on these two tags as well.

Reply:
Done